### ISSUE TITLE
What is the state of this project?

### ISSUE THREAD

**Issue by SmartDude1568** [zhuowei/ryujinx-ios#2]
This project has a lot of potential, and I love the idea of being able to play your Switch games on the go. (oops, just reinvented the Switch!)

If necessary, I could provide donations to help push this project forward, or I could help contribute with the limited knowledge of iOS that I have.

Thanks!

**Comment by duckcantcode** [comment]
I got a notification for this issue so I will go ahead and answer 😄 

It looks like in its current state the project is not usable, at least what I got from the CI won't run at all past the "Go!" menu. I asked and @.zhuowei is going to work on getting the project back to a usable state. I haven't figured out what the issue is yet that's preventing it from running.

The agenda once that's fixed is this: we still have the overarching issue of not having nearly enough virtual ram to run at console speeds. We can of course get as close as possible but the emulator might be a bit slow. We will also need access to JIT, which we have right now with AltStore/AltJIT/DirtyJIT/Jitterbug/20 others, but we're hoping iOS 17 might allow you to use JIT without having to turn it on with a computer program every time.

Long story short, we need to get the emulator back up and running, which means either [waiting for the dev to fix it](https://github.com/zhuowei/ryujinx-ios/issues/1#issuecomment-1466482819) , or building each commit and finding whichever the newest one that works is

**Comment by zhuowei** [comment]
> It looks like in its current state the project is not usable
The latest commit does work on my machine (tm). I don't have logging redirected, so error messages are only printed to Xcode console, and there's no way to install a firmware/keys, so you'd need to copy a Ryujinx folder from a macOS Ryujinx install. I don't think it's very easy to test this right now.

Right now it gets Pokemon: Brilliant Diamond to the language selection, then crashes with a Metal error when you select a language. Ryujinx on macOS does not do this, so I'm planning to trace the Metal calls to see why it's not working here.

>  iOS 17 might allow you to use JIT

I doubt it: sideloading requires all the code to be included in advance; you'd probably still need the same jit workarounds as before.

**Comment by duckcantcode** [comment]
> The latest commit does work on my machine (tm). I don't have logging redirected, so error messages are only printed to Xcode console, and there's no way to install a firmware/keys, so you'd need to copy a Ryujinx folder from a macOS Ryujinx install. I don't think it's very easy to test this right now.
> Right now it gets Pokemon: Brilliant Diamond to the language selection, then crashes with a Metal error when you select a language. Ryujinx on macOS does not do this, so I'm planning to trace the Metal calls to see why it's not working here.

Sorry, I misunderstood. Before I thought the GitHub Actions build wouldn't work because something was broken. I now think you are saying the build from Actions doesn't work but if you build the latest commit locally it will work? If that's the case, I'll build it and see if I can run homebrew or games with it.

Thank you very much for your work, by the way. The iOS emulation and sideloading scene has progressed so much because of your work.
